src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,69 +1,77 @@
-
-  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
-  blockName?: string;
-  /** Unique ID used by checkbox */
-  checkboxId?: string;
-  /** ClassName that appears in addition to "checkbox" */
-  className?: string;
-  /* For a controlled component only. If checked is not defined, the state of the uncontrolled native `input` component will be used. */
+  /** When using the Checkbox as a "controlled" form element, you can specify the Checkbox's checked state using this prop. Learn more about controlled and uncontrolled form fields: https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/ */
   checked?: boolean;
-  /* For uncontrolled component only. The initial value of an uncontrolled checkbox component. */
-  defaultChecked?: boolean;
-  /* The Label that the checkbox is using. */
-  labelOptions: LabelOptions;
+  /** className you can add in addition to 'input' */
+  className?: string;
+  /** Adds the 'disabled' prop to the input when true */
+  disabled?: boolean;
+  /** Helper for modifiers array; adds 'errored' styling */
+  errored?: boolean;
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
   /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
   modifiers?: string[];
-  /** name of the checkbox */
+  /** The name prop indicates into which group of checkboxes this checkbox belongs.  If none is specified, 'default' will be used */
   name?: string;
   /** The action to perform on the `<input>`'s onChange function  */
   onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** Offers the ability to show the checkbox's label onscreen or hide it. Refer to the `labelText` property for more information. */
+  showLabel: boolean;
+  /** Populates the value of the input */
+  value?: string;
 }
 
 const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
   (props, ref?) => {
     const {
-      name,
-      attributes,
-      modifiers = [],
-      className,
-      blockName = "",
-      checkboxId,
-      labelOptions,
       checked,
-      defaultChecked,
+      className,
+      disabled,
+      errored,
+      id,
+      labelText,
+      name,
       onChange,
+      showLabel,
+      value,
     } = props;
 
-    const baseClass = "checkbox";
+    const attributes = props.attributes || {};
+    const modifiers = props.modifiers ? props.modifiers : [];
+    const checkboxID = id || generateUUID();
+
+    if (!showLabel) attributes["aria-label"] = labelText;
 
+    if (errored) {
+      modifiers.push("error");
+    }
     return (
-      <div className={bem(baseClass, modifiers, blockName, [className])}>
+      <>
         <input
-          id={checkboxId}
-          name={name}
-          className={bem("input", [], baseClass)}
-          type="checkbox"
-          {...(checked !== undefined
```

Comment:
@kaplanmichael Did you intentionally remove this? The whole point of this conditional is to allow the component to be used a controlled component or uncontrolled component.

If you use pass a `checked` prop, then you're doing a "controlled" component, and it gives you `checked` prop as well as `aria-checked` and `onChange` props. If you don't pass a `checked` prop, then you're doing an "uncontrolled" component, and it just gives you a `defaultChecked` prop.

Reply:
You only need `aria-checked` if you are giving something other than a checkbox the the role of a checkbox (`role=checkbox`).  Also the user should be able to attach an onChange regardless of whether it is controlled or not.
For `defaultChecked`, you get the ability to query that regardless of whether it is passed as a prop or not (again, because we're using a native checkbox element and not some other element given the role of a checkbox).

Reply:
@nypl-wluisi this should be good now.  Please let me know.